File: Python/List-2/count_evens.py
import unittest

'''
Return the number of even ints in the given array. 
Note: the % "mod" operator computes the remainder, e.g. 5 % 2 is 1.


count_evens([2, 1, 2, 3, 4]) -> 3
count_evens([2, 2, 0]) -> 3
count_evens([1, 3, 5]) -> 0

'''

def count_evens(nums):
  
    count = 0
  
    for item in range(len(nums)):
    
        if nums[item]%2 == 0:
            count +=1
  
    
    return count  

class Test_count_evens(unittest.TestCase):

    def test_1(self):
        self.assertEqual(count_evens([2, 1, 2, 3, 4]), 3)

    def test_2(self):
        self.assertEqual(count_evens([2, 2, 0]), 3)

    def test_3(self):
        self.assertEqual(count_evens([1, 3, 5]), 0)


if __name__ == '__main__':
    unittest.main() 

File: Java/Array_2/countEvens.java
/**
 * @author Luis Teixeira
 *
 * @date   29/03/2018
 *
 * @mail   filipe.teixeira.996@gmail.com
 */
package Array_2;

import static org.junit.Assert.*;

import org.junit.Test;

/**
Return the number of even ints in the given array. Note: the % "mod" operator computes the remainder, e.g. 5 % 2 is 1.


countEvens([2, 1, 2, 3, 4]) --> 3
countEvens([2, 2, 0]) --> 3
countEvens([1, 3, 5]) --> 0
 */
public class countEvens {

	private static int i = 1;
	
	@Test
	public void Test1()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("countEvens([2, 1, 2, 3, 4]) --> 3\n");
		int[] numA = {2, 1, 2, 3, 4};
		assertEquals("countEvens([2, 1, 2, 3, 4]) --> 3", 3, countEvens(numA));
		i++;
	}
	
	@Test
	public void Test2()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("countEvens([2, 2, 0]) --> 3\n");
		int[] numA = {2, 2, 0};
		assertEquals("countEvens([2, 2, 0]) --> 3", 3, countEvens(numA));
		i++;
	}
	
	@Test
	public void Test3()
	{
		System.out.println("**********Test "+i+"**********");
		System.out.println("countEvens([1, 3, 5]) --> 0\n");
		int[] numA = {1, 3, 5};
		assertEquals("countEvens([1, 3, 5]) --> 0", 0, countEvens(numA));
		i++;
	}
	
	public int countEvens(int[] nums) {
		
		int count = 0;
		
		for (int i = 0; i < nums.length; i++) {
			if (nums[i] % 2 == 0) {
				count++;
			}
		}
		
		return count;
	}
}
